lib/usb/usb_standard.c

```diff
@@ -247,19 +248,40 @@ static int usb_standard_set_configuration(usbd_device *usbd_dev,
 					  struct usb_setup_data *req,
 					  uint8_t **buf, uint16_t *len)
 {
-	int i;
+	uint8_t i;
+	const struct usb_config_descriptor *cfg;
 
 	(void)req;
 	(void)buf;
 	(void)len;
 
-	/* Is this correct, or should we reset alternate settings. */
-	if (req->wValue == usbd_dev->current_config) {
-		return 1;
+	if(req->wValue > 0) {
+		if (req->wValue > usbd_dev->desc->bNumConfigurations) {
```

Comment:
yes, i know that. when i raised the same issue, i was told that they need to be sequencial and must start from 1.
same design problem for interface-number, interface-alternate-setting.